operator/v1/types_console.go

```diff
@@ -143,12 +143,32 @@ type ProjectAccess struct {
 	AvailableClusterRoles []string `json:"availableClusterRoles,omitempty"`
 }
 
+// If sub-catalog type(s) is(are) mentioned in both `enabled` and `disabled` list, the sub-catalog type(s) should be hidden.
+type DeveloperConsoleCatalogTypes struct {
+	// enabled is a list of developer catalog types (sub-catalogs IDs) that will be shown to users.
+	// Types (sub-catalogs) are added via console plugins, the available types (sub-catalog IDs) are available
+	// in the console on the cluster configuration page, or when editing the YAML in the console.
+	// A new type will not be shown to the user until it is added in the enabled list.
+	// If `enabled` list is empty, complete developer catalog should be hidden.
+	// +optional
+	Enabled []string `json:"enabled"`
+	// disabled is a list of developer catalog types (sub-catalogs IDs) that are not shown to users.
+	// Types (sub-catalogs) are added via console plugins, the available types (sub-catalog IDs) are available
+	// in the console on the cluster configuration page, or when editing the YAML in the console.
+	// If all the available sub-catalog types are added in the `disabled` list, complete developer catalog should be hidden.
```

Comment:
With enabled, there's a way to enable all (this is the default when neither list is configured). We then have a way to enable only certain items, and to disable only certain items.

There's currently no way to permanently disable all items (new catalogs could be added but not in the list).

I wonder if we need a wildcard option, let me see how others tackle this

Reply:
> I wonder if we need a wildcard option, let me see how others tackle this

Interesting observation.  Kube admission configuration is often used as an example that allows a default list with default on and default off as an option.  Then configuration values that forceEnable and forceDisable.  Very much like (exactly like?) this API.  But we didn't consider the case of, "disable everything".  I'm open to a wildcard value like "*".

Reply:
There is an alternative suggestion to this API as well, which is to make a discriminated union which would look something like below:
```
state: All|Enabled|Disabled|None
enabled: []
disabled: []
```

Though I'm concerned about the 4 values of the enum, I'm not sure it's intuitive that All and Enabled are different from one another, likewise with Disabled and None.

Perhaps if the list was All, Explicit and None and then the structure would be
```
state: All|Explicit|None
explicit:
  enabled: []
  disabled: []
```
I think I prefer the current approach to the DU approach personally

Reply:
We're fine with both, please let us know which direction we should follow...

Reply:
In this discriminated union:
* Default state to Enabled for backwards compatibility 
* Enabled + empty enabled[]: show everything
* Enabled + non-empty enabled[]: show only the items in the enabled list
* Disabled + empty disabled[]: disable everything
* Disabled + non-empty disabled[]: disable only the items in the disabled list
```
state: Enabled|Disabled
enabled: []
disabled: []
```
This means that you cannot use enabled and disabled at the same time, which, as we don't have specific lists of enabled and disabled by default (default is all on), we don't need to use them at the same time